src/components/Alert.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { FunctionComponent, ReactNode, useMemo } from 'react'
+import styled, { css } from 'styled-components'
+import SvgIcon from '~/shared/components/SvgIcon'
+import { COLORS, MEDIUM } from '~/shared/utils/styled'
+import Spinner from '~/shared/components/Spinner'
+
+type AlertType = 'loading' | 'error' | 'success' | 'notice'
+export const Alert: FunctionComponent<{
+    type: AlertType
+    title: string
+    children: ReactNode
+}> = ({ type, title, children }) => {
+    const icon: ReactNode = useMemo(() => {
+        switch (type) {
+            case 'success':
+                return <StyledIcon name="checkmark" />
+            case 'error':
+                return <StyledIcon name="warnBadge" />
+            case 'loading':
+                return (
+                    <SpinnerWrapper>
+                        <Spinner color="blue" />
+                    </SpinnerWrapper>
+                )
+            case 'notice':
+                return <StyledIcon name="infoBadge" />
+        }
+    }, [type])
+
+    return (
+        <AlertWrap $type={type}>
+            {icon}
+            <div>
+                <Title>{title}</Title>
+                {children}
+            </div>
+        </AlertWrap>
+    )
+}
+
+const AlertWrap = styled.div<{ $type: AlertType }>`
+    display: flex;
+    gap: 16px;
+    padding: 20px 16px;
+    border-radius: 8px;
+  
+  ${({ $type }) => {
+      let color = 'transparent'
+      switch ($type) {
+          case 'success':
+              color = COLORS.alertSuccessBackground
+              break
+          case 'error':
+              color = COLORS.alertErrorBackground
+              break
+          case 'loading':
+          case 'notice':
+              color = COLORS.alertInfoBackground
+              break
+      }
+      return css`
+          background-color: ${color};
+      `
+  }}}
+`
```

Comment:
For readability this could get extracted into a local getter, e.g.

```ts
getColorForAlertType({ $type }: { $type?: AlertType }): string {
    // logic logic logic
}
```

Then

```tsx
styled.div`
    color: ${getColorForAlertType($type)};
`
```

Reply:
Updated the snippets a bit, FYI.